### PR TITLE
Consolidate read/write methods in `sessions` modules to the `session.Session` parent class.

### PR DESCRIPTION
**Pull request by dougthor42**
This is some pre-work for #72. I noticed that there was a fair amount of code duplication in the various `sessions` modules, so I cleaned that up by moving any pure duplicates to the `session.Session` parent class.

I also move some common USB code to a `BaseUSBSession` class.

The only one unchanged was `serial.py` as those read/write methods are different. I haven't investigated if the _need_ to be different though.

Hopefully test coverage is sufficient to ensure that this doesn't break things. We have line coverage for the `write` methods, but are missing a couple lines in `read`.

### Comments and Open Questions

1.  I noticed that `mypy` is failing for channels.py:199. I figured that could be fixed in a separate PR. LMK if it needs to be fixed before merging this.
2.  Do you want me to add anything to `CHANGES.rst?

### CODE REVIEW

File: pyvisa_sim/sessions/usb.py [hunk 1]
```diff
@@ -32,104 +30,12 @@ def after_parsing(self) -> None:
...
-            self.parsed.board
-        )
-
-    def read(self, count: int) -> Tuple[bytes, constants.StatusCode]:
-        end_char, _ = self.get_attribute(constants.ResourceAttribute.termchar)
-        enabled, _ = self.get_attribute(constants.ResourceAttribute.termchar_enabled)
-        timeout, _ = self.get_attribute(constants.ResourceAttribute.timeout_value)
-        timeout /= 1000
-
-        now = start = time.time()
```
Comment (dougthor42) [inline pyvisa_sim/sessions/usb.py:102]: This is technically a minor code difference from the other `read` methods, but the _functionality_ of the code is exactly the same.

**Comment by MatthieuDartiailh** [comment]
If you have the bandwidth to do so I would be very keen on you fixing mypy.

I will try to review ASAP.

**Comment by MatthieuDartiailh** [comment]
Once you rebase on latest main I will re-check and merge.

**Comment by MatthieuDartiailh** [comment]
Actually thinking a tiny bit more about this, there is one minor issue. 

Technically not all sessions support read and write (register based devices). We do not support them yet (and we may never do so) but to be a tad more future proof I would feel better extracting the logic in a MessageBasedSession that could then be inherited by tcpip, usb, gpib and serial resources. Would you mind making such a change ?

**Review by MatthieuDartiailh — APPROVED** [review]

**Pull request merged**